src/sentry_logger.cpp

```diff
@@ -24,15 +26,15 @@ const char *error_types[] = {
 	"USER SHADER ERROR"
 };
 
-SentryLogger::ErrorType error_type_as_enum[] = {
-	SentryLogger::ERROR_TYPE_ERROR,
-	SentryLogger::ERROR_TYPE_WARNING,
-	SentryLogger::ERROR_TYPE_SCRIPT,
-	SentryLogger::ERROR_TYPE_SHADER,
-	SentryLogger::ERROR_TYPE_ERROR,
-	SentryLogger::ERROR_TYPE_WARNING,
-	SentryLogger::ERROR_TYPE_SCRIPT,
-	SentryLogger::ERROR_TYPE_SHADER,
+GodotErrorType error_type_as_enum[] = {
+	GodotErrorType::ERROR_TYPE_ERROR,
+	GodotErrorType::ERROR_TYPE_WARNING,
+	GodotErrorType::ERROR_TYPE_SCRIPT,
+	GodotErrorType::ERROR_TYPE_SHADER,
+	GodotErrorType::ERROR_TYPE_ERROR,
```

Comment:
Are the duplicates intentional?

Reply:
I mean the individual entries in the enum are duplicated.

Reply:
Yes, and this is intentional. You need to look at both definitions at the same time:
```cpp
const char *error_types[] = {
	"ERROR",
	"WARNING",
	"SCRIPT ERROR",
	"SHADER ERROR",
	"USER ERROR",
	"USER WARNING",
	"USER SCRIPT ERROR",
	"USER SHADER ERROR"
};

GodotErrorType error_type_as_enum[] = {
	GodotErrorType::ERROR_TYPE_ERROR,
	GodotErrorType::ERROR_TYPE_WARNING,
	GodotErrorType::ERROR_TYPE_SCRIPT,
	GodotErrorType::ERROR_TYPE_SHADER,
	GodotErrorType::ERROR_TYPE_ERROR,
	GodotErrorType::ERROR_TYPE_WARNING,
	GodotErrorType::ERROR_TYPE_SCRIPT,
	GodotErrorType::ERROR_TYPE_SHADER,
};
```
The first array are the strings that we look for in the log file. And the second array are the error types we map to from the values of the first array. Notice that both arrays have the same number of elements. 
So `error_types[0] => error_type_as_enum[0];` and so on.
It's just a faster mapping than a hash map would be. I'll add comments to the code.